composer/core/evaluator.py

```diff
@@ -51,17 +48,25 @@ def should_eval(state: State, event: Event):
         if eval_at_fit_end and event == Event.FIT_END and state.timestamp.batch != last_batch_seen:
             return True
 
-        if eval_interval.unit == TimeUnit.EPOCH and int(
-                state.timestamp.epoch) % int(eval_interval) == 0 and event == Event.EPOCH_END:
+        assert state.previous_timestamp is not None
```

Comment:
Return false if None? Otherwise, if algo calls before it's set, you get nasty trace

Reply:
same answer, will do though.